docs/maintaining.md

```diff
@@ -92,3 +92,118 @@ made to always install the latest released version.
 
 Update the release branch by fast-forwarding it to the appropriate SHA (matching
 the latest tag) and pushing it back upstream.
+
+## `grafana/prometheus` Maintainence
+
+Grafana Labs is using the Agent for their internal monitoring and want to take
+advantage of the Agent to proof-of-concept additions to Prometheus before they
+get moved upstream. A `grafana/prometheus` repository is maintained by Grafana
+Labs where all non-trivial changes will go first. Doing this allows for getting
+Cortex-specific changes moving along faster and providing strong evidence
+towards its usefulness and correctness when it becomes part of an upstream PR.
+
+We are commiting ourselves to doing the following:
+
+1. Always use a recent Prometheus release: the Agent will always vendor a
+   recent Prometheus release and not Prometheus master. We want the Agent's
+   Prometeheus roots to be stable.
+2. Keep changes mergeable upstream: we want to continue to be good OSS citizens,
+   and we intend that all features we add to our Prometheus repository will
+   become an upstream PR. We will maintain our repository in a way that supports
+   doing this.
+3. Reduce code drift: The code the Agent uses on top of Prometheus will be
+   layered on top of a Prometheus release rather than sandwiched in between.
+
+Maintainence of the `grafana/prometheus` repository revolves around feature
+branches (named `feat-SOME-FEATURE`) and release branches (named
+`release-vX.Y.Z-grafana`). The release branches will always use the same release
+version as the `prometheus/prometheus` release it is based off of.
+
+By adding features to the `grafana/prometheus` repository first, we are
+committing ourselves to extra maintenance of features that have not yet been
+merged upstream. Feature authors will have to babysit their features to
+coordinate with the Prometheus release schedule to always be compatible. One the
+feature is merged upstream, this burden of maintaining is made easier as the
+Prometheus team can more easily sync breaking changes across the whole
+repository.
+
+We are purposefully carrying this extra burden because we intend to ultimately
+make Prometheus better and contribute all of our enhancements upstream. We want
+to strive to benefit the Prometheus ecosystem at large.
+
+### Creating a New Feature
+
+For `grafana/prometheus` maintainers to create a new feature, they will do the
+following:
+
+1. Create a feature branch in `grafana/prometheus` based on the latest release
+   tag that `grafana/prometheus` currently has a release branch for. The feature
+   branch should follow the naming convention `feat-<feature name>`.
+2. Implement the feature and open a PR to merge the feature branch into the
+   associated `grafana/prometheus` release branch.
+3. Once the release branch is updated, open a PR to update `grafana/agent` to
```

Comment:
How do you feel about listing the current version we're using somewhere? That way we can keep track of which branch should we currently using. The alternative is scanning through PRs to check which is latest as I _think_ go.mod won't list it in a human-readable format, right?

Reply:
The main idea behind my comment is to eventually add: `to know which is the current branch and version we're tracking please refer to this link in the readme somewhere`

Reply:
This would be a good thing to include in the README along with our reasoning for not using `prometheus/prometheus` directly.